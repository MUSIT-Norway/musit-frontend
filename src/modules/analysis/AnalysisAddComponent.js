/* @flow */
import React, { PropTypes } from 'react';
import { I18n } from 'react-i18nify';
import {
  Radio,
  PageHeader,
  Form,
  FormGroup,
  Col,
  FormControl,
  Button,
  Well,
  Panel
} from 'react-bootstrap';
import FontAwesome from 'react-fontawesome';
import { SaveCancel } from '../../components/formfields/index';
import { hashHistory } from 'react-router';
import Config from '../../config';
import type { AppSession } from '../../types/appSession';
import type { FormData, Update } from './types/form';
import type { ObjectData } from '../../types/object';
import Label from './components/Label';
import FieldGroup from './components/FIeldGroup';
import AddButton from './components/AddButton';
import NewLine from './components/NewLine';

const { Table } = require('reactable');

type Location = {
  state?: Array<ObjectData>
};

type AnalysisType = { id: number, name: string };

type Store = {
  analysisTypes: AnalysisType[]
};

type Props = {
  form: FormData,
  updateForm: Update,
  store: Store,
  appSession: AppSession,
  saveAnalysisEvent: Function,
  location: Location
};

const getValue = field => field.rawValue || '';

export const saveAnalysisEventLocal = (
  appSession: AppSession,
  form: FormData,
  location?: Location,
  saveAnalysisEvent: Function
) =>
  saveAnalysisEvent({
    museumId: appSession.museumId,
    data: {
      analysisTypeId: getValue(form.analysisTypeId),
      eventDate: getValue(form.registeredDate),
      note: getValue(form.note),
      objectIds: location && location.state ? location.state.map(a => a.uuid) : [],
      result: {
        by: getValue(form.by),
        expirationDate: getValue(form.expirationDate),
        reason: getValue(form.reason),
        caseNumbers: getValue(form.caseNumbers),
        cancelledBy: getValue(form.cancelledBy),
        cancelledReason: getValue(form.cancelledReason)
      }
    },
    token: appSession.accessToken
  });

const updateFormField = (field, updateForm) =>
  e =>
    updateForm({
      name: field.name,
      rawValue: e.target.value
    });

export const goToAnalysis = (
  fn: Promise<string>,
  appSession: AppSession,
  goTo: Function = hashHistory.push
) =>
  fn.then((analysisId: string) =>
    goTo(Config.magasin.urls.client.analysis.viewAnalysis(appSession, analysisId)));

const AnalysisAdd = (
  { form, updateForm, store, saveAnalysisEvent, appSession, location }: Props
) => (
  <div>
    <br />
    <PageHeader style={{ paddingLeft: 20 }}>
      {I18n.t('musit.analysis.registeringAnalysis')}
    </PageHeader>
    <Col md={12}>
      <strong>HID:</strong>{' '}{getValue(form.id)}
    </Col>
    <Col md={12}>
      <strong>Registrert:</strong>
      {' '}
      <FontAwesome name="user" />
      {' '}
      {getValue(form.registeredBy)}
      {' '}
      <FontAwesome name="clock-o" />{' '}{getValue(form.registeredDate)}
    </Col>
    <Col md={12}>
      <strong>Sist endret:</strong>
      {' '}
      <FontAwesome name="user" />
      {' '}
      {getValue(form.doneBy)}
      {' '}
      <FontAwesome name="clock-o" />
      {' '}
      {getValue(form.doneDate)}
      {' '}
      <Button bsStyle="link">Se endringshistorikk</Button>
    </Col>
    <NewLine />
    <Form>
      <FormGroup>
        <FieldGroup
          id="formControlsText"
          type="text"
          label="Saksnummer"
          value={getValue(form.caseNumbers)}
          onChange={updateFormField(form.caseNumbers, updateForm)}
        />
      </FormGroup>
      <FormGroup>
        <AddButton id="1" label="Legg til saksnummer" md={5} />
      </FormGroup>
    </Form>
    <NewLine />
    <Form inline>
      <Col md={12}><h5><strong>Objekt/prøve</strong></h5></Col>
      <Col mdOffset={1} md={5}>
        <Table
          className="table"
          columns={[
            { key: 'museumNo', label: 'Museumsnr' },
            { key: 'subNo', label: 'Unr' },
            { key: 'term', label: 'Term/artsnavn' }
          ]}
          data={location ? location.state || [] : []}
          sortable={['museumNumber', 'subNumber', 'term']}
          noDataText="Ingen objekter"
        />
      </Col>
      <AddButton id="2" label="Legg til objekt" md={11} mdOffset={1} />
    </Form>
    <NewLine />
    <Form horizontal style={{ paddingLeft: 20 }}>
      <FormGroup>
        <Col md={12}><h5><strong>Personer tilknyttet analysen</strong></h5></Col>
      </FormGroup>
      <FormGroup>
        <FieldGroup
          id="responsible"
          type="text"
          label="Navn"
          placeholder="Fornavn Etternavn"
          value={getValue(form.responsible)}
          onChange={updateFormField(form.responsible, updateForm)}
        />
        <Label label="Rolle" md={1} />
        <Col md={1}>
          <FormControl componentClass="select" placeholder="Velg rolle">
            <option value="Velgsted">Velg rolle</option>
            <option value="other">...</option>
          </FormControl>
        </Col>
        <AddButton id="3" label="Legg til person" md={2} />
      </FormGroup>
    </Form>
    <NewLine />
    <FormGroup>
      <Label label="Analysested" md={1} />
      <Col md={2}>
        <FormControl componentClass="select" placeholder="Velg sted">
          <option value="Velgsted">Velg sted</option>
          <option value="other">...</option>
        </FormControl>
      </Col>
    </FormGroup>
    <NewLine />
    <Well>
      <Form horizontal>
        <FormGroup>
          <Label label="Type analyse" md={1} />
          <Col md={2}>
            <FormControl
              componentClass="select"
              placeholder="Velg kategori"
              onChange={updateFormField(form.analysisTypeId, updateForm)}
            >
              <option>Velg kategori</option>
              {store.analysisTypes.map(a => (
                <option key={a.id} value={a.id}>{a.name}</option>
              ))}
            </FormControl>
          </Col>
        </FormGroup>
        <FormGroup>
          <FieldGroup
            id="formControlsText"
            type="text"
            label="Ekstern kilde"
            placeholder="http://www.lenke.no"
            value={getValue(form.externalSource)}
            onChange={updateFormField(form.externalSource, updateForm)}
          />
          <Col md={2}>
            <Button>Lagre</Button>
          </Col>
        </FormGroup>
        <FormGroup>
          <FieldGroup id="formControlsText" type="text" label="Last opp fil" />
          <Col md={2}>
            <Button>Bla gjennom</Button>
          </Col>
        </FormGroup>
        <FormGroup>
          <Label label="Kommentar / resultat" md={1} />
          <Col md={5}>
            <FormControl
              componentClass="textarea"
              placeholder=""
              value={getValue(form.comments)}
              onChange={updateFormField(form.comments, updateForm)}
            />
          </Col>
        </FormGroup>
        <FormGroup>
          <Label label="Klausulering" md={1} />
          <Col md={5}>
            <Radio checked readOnly inline>
              Ja
            </Radio>
            <Radio inline readOnly>
              Nei
            </Radio>
          </Col>
        </FormGroup>
        <FormGroup>
          <Panel
            collapsible
            expanded
            style={{ border: 'none', backgroundColor: '#f5f5f5' }}
          >
            <FormGroup>
              <FieldGroup
                id="navn"
                md={1}
                type="text"
                label="Klausulert for"
                placeholder="Fornavn Etternavn"
                value={getValue(form.by)}
                onChange={updateFormField(form.by, updateForm)}
              />
            </FormGroup>
            <FormGroup>
              <FieldGroup
                id="navn"
                md={1}
                type="text"
                label="Årsak til klausulering"
                value={getValue(form.reason)}
                onChange={updateFormField(form.reason, updateForm)}
              />
            </FormGroup>
            <FormGroup>
              <FieldGroup
                id="Saksnummer"
                md={1}
                type="text"
                label="Saksnummer"
                value={getValue(form.caseNumbers)}
                onChange={updateFormField(form.caseNumbers, updateForm)}
              />
            </FormGroup>
            <FormGroup>
              <AddButton id="3" label="Legg til flere saksnummer" md={11} mdOffset={1} />
            </FormGroup>
            <FormGroup>
              <FieldGroup
                id="navn"
                md={1}
                type="text"
                label="Sluttdato"
                value={getValue(form.expirationDate)}
                readOnly
              />
            </FormGroup>
            <FormGroup>
              <FieldGroup
                id="navn"
                md={1}
                type="text"
                label="Opphevet av"
                placeholder="Fornavn Etternavn"
                value={getValue(form.cancelledBy)}
                onChange={updateFormField(form.cancelledBy, updateForm)}
              />
            </FormGroup>
            <FormGroup>
              <FieldGroup
                id="navn"
                md={1}
                type="text"
                label="Årsak til oppheving"
                placeholder=""
                value={getValue(form.cancelledReason)}
                onChange={updateFormField(form.cancelledReason, updateForm)}
              />
            </FormGroup>
          </Panel>
        </FormGroup>
      </Form>
    </Well>
    <Form horizontal style={{ paddingLeft: 20 }}>
      <FormGroup
        controlId={form.note.name}
        validationState={form.note.status && !form.note.status.valid ? 'error' : null}
      >
        <Label label="Kommentar til analysen" md={1} />
        <Col md={5}>
          <FormControl
            className="note"
            onChange={updateFormField(form.note, updateForm)}
            componentClass="textarea"
            placeholder={form.note.name}
            value={getValue(form.note)}
          />
        </Col>
      </FormGroup>
      <FormGroup>
        <Label label="Avslutt analyse" md={1} />
        <Col md={5}>
          <Radio inline readOnly>
            Ja
          </Radio>
          <Radio inline checked readOnly>
            Nei
          </Radio>
        </Col>
      </FormGroup>
    </Form>
    <NewLine />
    <SaveCancel
      onClickSave={() =>
        goToAnalysis(
          saveAnalysisEventLocal(appSession, form, location, saveAnalysisEvent),
          appSession
        )}
    />
    <NewLine />
    <Form horizontal>
      <FormGroup>
        <Col mdOffset={1}><h5><strong>Endringshistorikk</strong></h5></Col>
      </FormGroup>
      <FormGroup>
        <Col mdOffset={1}>
          {getValue(form.registeredBy)} - {getValue(form.registeredDate)}
        </Col>
      </FormGroup>
      <FormGroup>
        <Col mdOffset={1}>{getValue(form.doneBy)} - {getValue(form.doneDate)}</Col>
      </FormGroup>
      <FormGroup>
        <Col mdOffset={1}><Button bsStyle="link">Se mer</Button></Col>
      </FormGroup>
    </Form>
  </div>
);

const FieldShape = {
  name: PropTypes.string.isRequired,
  rawValue: PropTypes.string,
  status: PropTypes.shape({
    valid: PropTypes.bool,
    error: PropTypes.any
  })
};

const FieldShapeBoolean = {
  name: PropTypes.string.isRequired,
  rawValue: PropTypes.bool,
  status: PropTypes.shape({
    valid: PropTypes.bool,
    error: PropTypes.any
  })
};

AnalysisAdd.propTypes = {
  form: PropTypes.shape({
    id: PropTypes.shape(FieldShape).isRequired,
    analysisTypeId: PropTypes.shape(FieldShape).isRequired,
    doneBy: PropTypes.shape(FieldShape).isRequired,
    doneDate: PropTypes.shape(FieldShape).isRequired,
    registeredBy: PropTypes.shape(FieldShape).isRequired,
    registeredDate: PropTypes.shape(FieldShape).isRequired,

    responsible: PropTypes.shape(FieldShape).isRequired,

    administrator: PropTypes.shape(FieldShape).isRequired,
    completedBy: PropTypes.shape(FieldShape).isRequired,
    completedDate: PropTypes.shape(FieldShape).isRequired,
    objectId: PropTypes.shape(FieldShape).isRequired,
    note: PropTypes.shape(FieldShape).isRequired,
    type: PropTypes.shape(FieldShape),

    partOf: PropTypes.shape(FieldShape).isRequired,
    result: PropTypes.shape(FieldShape).isRequired,
    place: PropTypes.shape(FieldShape).isRequired,

    externalSource: PropTypes.shape(FieldShape).isRequired,
    comments: PropTypes.shape(FieldShape).isRequired,

    restrictions: PropTypes.shape(FieldShapeBoolean).isRequired,
    by: PropTypes.shape(FieldShape).isRequired,
    expirationDate: PropTypes.shape(FieldShape).isRequired,
    reason: PropTypes.shape(FieldShape).isRequired,
    caseNumbers: PropTypes.shape(FieldShape).isRequired,
    cancelledBy: PropTypes.shape(FieldShape).isRequired,
    cancelledReason: PropTypes.shape(FieldShape).isRequired,

    completeAnalysis: PropTypes.shape(FieldShapeBoolean).isRequired,
    museumNo: PropTypes.shape(FieldShape).isRequired,
    term: PropTypes.shape(FieldShape).isRequired
  }).isRequired,
  updateForm: PropTypes.func.isRequired,
  store: PropTypes.object.isRequired,
  saveAnalysisEvent: PropTypes.func
};

export default AnalysisAdd;
