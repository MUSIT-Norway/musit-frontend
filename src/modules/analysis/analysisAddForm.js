/* @flow */
import type { Field } from '../../forms/form';
import createForm from '../../forms/form';
import {
  composeValidators,
  isNumber,
  isRequired,
  isNumberInRange //,
//  isSpecialPhone
} from '../../forms/validators';

// import type {
//   Phone
// } from '../../forms/types';
import {
  stringMapper,
  numberMapper,
  booleanMapper//,
  // specialPhoneMapper
} from '../../forms/mappers';


const getStrField = (field, value = ''): Field<string> => {
  return  {
    name: field,
    mapper: stringMapper,
    defaultValue: value,
    validator: {
      rawValidator: isRequired
    }
  };
};

const getBoolField = (field, value = false): Field<boolean> => {
  return  {
    name: field,
    defaultValue: value,
    mapper: booleanMapper,
    validator: {
    }
  };
};

const id = getStrField('id', '123');
const registeredBy = getStrField('registeredBy', 'Per Hansen');
const registeredDate = getStrField('registeredDate', '12.12.2012');
const doneBy = getStrField('doneBy', 'Per Hansen');
const doneDate = getStrField('doneDate', '12.12.2012');
const eventDate = getStrField('eventDate', '13.12.2012');
const objectId = getStrField('objectId', '1234456');
const partOf = getStrField('partOf', '2345');
const result = getStrField('result', 'result');
const actor = getStrField('actor', 'Test Name Actor');
const role = getStrField('role', 'Creator');
const place = getStrField('place', 'Oslo, Norway');
const analysisTypeId = getStrField('analysisTypeId', 'b15ee459-38c9-414f-8b54-7c6439b44d3d');
const externalSource = getStrField('externalSource', 'www.musit.uio.no');
const comments = getStrField('comments', 'comments');
const restrictions = getBoolField('restrictions', true);
const restrictionsFor = getStrField('restrictionsFor', 'Test name');
const reasonForRestrictions = getStrField('reasonForRestrictions', 'Reason for Restrictions');
const restrictionsEndDate = getStrField('restrictionsEndDate', '13.12.2012');
const repealedBy = getStrField('repealedBy', 'repealed By Person Name');
const note = getStrField('note', 'default value of note');
const completeAnalysis = getBoolField('completeAnalysis', true);

const caseNumber: Field<number> = {
  name: 'caseNumber',
  mapper: numberMapper,
  defaultValue: 1234,
  validator: {
    rawValidator: composeValidators(isRequired, isNumber(0, 2)),
    valueValidator: isNumberInRange(0, 10)
  }
};


const fieldsArray = [
  id,
  registeredBy,
  registeredDate,
  doneBy,
  doneDate,
  eventDate,
  objectId,
  partOf,
  result,
  caseNumber,
  actor,
  role,
  place,
  analysisTypeId,
  externalSource,
  comments,
  restrictions,
  restrictionsFor,
  reasonForRestrictions,
  restrictionsEndDate,
  repealedBy,
  note,
  completeAnalysis
];
export default createForm('analysisAddForm.js', fieldsArray);


