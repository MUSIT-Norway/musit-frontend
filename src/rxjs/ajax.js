import {Observable} from 'rxjs';
import 'rxjs/add/observable/dom/ajax';

export const onComplete = (callback) => (response) => {
  if (callback && callback.onComplete) {
    callback.onComplete(response);
  }
};

export const onFailure = (callback) => (error) => {
  if (callback && callback.onFailure) {
    callback.onFailure(error);
  }
  return Observable.of((state) => ({...state, error}));
};

export const toResponse = ({ response }) => response;

const ajax = (url, method, body, token, headers) =>
  Observable.ajax({
    url,
    responseType: 'json',
    method,
    body,
    headers: {
      'Authorization': 'Bearer ' + token,
      ...headers
    }
  });

const simpleAjax = (ajax$, callback) =>
  ajax$.map(toResponse)
    .do(onComplete(callback))
    .catch(onFailure(callback));

export const get = (url, token) => ajax(url, 'GET', null, token);

export const simpleGet = (url, token, callback) => simpleAjax(get(url, token), callback);

export const del = (url, token) => ajax(url, 'DELETE', null, token);

export const simpleDel = (url, token, callback) => simpleAjax(del(url, token), callback);

export const post = (url, body, token) => ajax(url, 'POST', body, token, { 'Content-Type': 'application/json' });

export const simplePost = (url, data, token, callback) => simpleAjax(post(url, data, token), callback);

export const put = (url, data, token) => ajax(url, 'PUT', data, token, { 'Content-Type': 'application/json' });

export const simplePut = (url, data, token, callback) => simpleAjax(put(url, data, token), callback);