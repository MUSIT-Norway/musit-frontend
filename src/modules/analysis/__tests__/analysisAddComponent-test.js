import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import AnalysisAddComponent, {
  saveAnalysisEventLocal,
  goToAnalysis
} from '../AnalysisAddComponent';
import { fieldsArray } from '../analysisAddForm';
import sinon from 'sinon';
import { Observable } from 'rxjs';

const objectsData = [
  {
    museumNumber: '123',
    subNumber: '12345678911',
    term: 'Spyd',
    uuid: '1cbf15cb-8348-4e66-99a4-bc314da57a42'
  },
  {
    museumNumber: '124',
    subNumber: '12345678912',
    term: 'Beltering',
    uuid: '2cbf15cb-8348-4e66-99a4-bc314da57a42'
  },
  {
    museumNumber: '125',
    subNumber: '12345678913',
    term: 'Ã˜sekar',
    uuid: '3cbf15cb-8348-4e66-99a4-bc314da57a42'
  }
];

const analysisTypes = [
  {
    category: 8,
    id: '1bbf15cb-8348-4e66-99a4-bc314da57a42',
    name: '3D-skanning, laser'
  },
  {
    category: 8,
    id: 'b39399ab-aabd-4ebc-903b-adcf6876a364',
    name: '3D-skanning, strukturert lys'
  }
];

const store = {
  analysisTypes: analysisTypes,
  objectsData: objectsData
};

const form = fieldsArray.reduce(
  (acc, n) => ({
    ...acc,
    [n.name]: {
      name: n.name,
      rawValue: n.name === 'note' ? 'test note' : n.mapper.toRaw(n.defaultValue)
    }
  }),
  {}
);

const appSession = {
  museumId: 99,
  accessToken: '1234'
};

describe('AnalysisAddComponent', () => {
  it('saveAnalysisEventLocal should call saveAnalysisEvent', () => {
    const saveAnalysisEvent = sinon.spy();
    saveAnalysisEventLocal(appSession, form, store, saveAnalysisEvent)(1);
    expect(saveAnalysisEvent.calledOnce).toBe(true);
    expect(saveAnalysisEvent.getCall(0).args[0].museumId).toBe(99);
    expect(saveAnalysisEvent.getCall(0).args[0].token).toBe('1234');
    expect(saveAnalysisEvent.getCall(0).args[0].data.analysisTypeId).toBe(
      'b15ee459-38c9-414f-8b54-7c6439b44d3d'
    );
  });

  it('Call goToAnalysis.', done => {
    const fakeGoTo = sinon.spy();
    const fakeFn = () => Observable.of(null).toPromise();
    const fn = goToAnalysis(fakeFn, appSession, fakeGoTo);
    const analysisId = 2;
    fn(analysisId).then(() => {
      expect(fakeGoTo.calledOnce).toBe(true);
      done();
    });
  });

  it('should fire updateForm when input is changing', () => {
    const updateForm = sinon.spy();
    const wrapper = mount(
      <AnalysisAddComponent form={form} updateForm={updateForm} store={store} />
    );
    wrapper.find('.note').simulate('change', {
      target: {
        value: 'note changed'
      }
    });
    expect(updateForm.getCall(0).args[0].name).toBe('note');
    expect(updateForm.getCall(0).args[0].rawValue).toBe('note changed');
  });

  it('should render properly', () => {
    const updateForm = sinon.spy();
    const wrapper = shallow(
      <AnalysisAddComponent form={form} updateForm={updateForm} store={store} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
