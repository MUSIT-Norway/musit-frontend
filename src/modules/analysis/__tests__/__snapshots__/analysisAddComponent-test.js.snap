exports[`AnalysisAddComponent should render properly 1`] = `
<div>
  <br />
  <PageHeader
    bsClass="page-header"
    style={
      Object {
        "paddingLeft": 20
      }
    }>
    Registrer analyse
  </PageHeader>
  <Col
    bsClass="col"
    componentClass="div"
    md={12}>
    <strong>
      HID:
    </strong>
     
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    md={12}>
    <strong>
      Registrert:
    </strong>
     
    <FontAwesome
      name="user" />
     
     
    <FontAwesome
      name="clock-o" />
     
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    md={12}>
    <strong>
      Sist endret:
    </strong>
     
    <FontAwesome
      name="user" />
     
     
    <FontAwesome
      name="clock-o" />
     
     
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="link"
      disabled={false}>
      Se endringshistorikk
    </Button>
  </Col>
  <NewLine />
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={false}
    inline={false}>
    <FormGroup
      bsClass="form-group">
      <FieldGroup
        id="formControlsText"
        label="Saksnummer"
        md={1}
        onChange={[Function anonymous]}
        type="text"
        value={Array []} />
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <AddButton
        id="1"
        label="Legg til saksnummer"
        md={5}
        mdOffset={0} />
    </FormGroup>
  </Form>
  <NewLine />
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={false}
    inline={true}>
    <Col
      bsClass="col"
      componentClass="div"
      md={12}>
      <h5>
        <strong>
          Objekt/prøve
        </strong>
      </h5>
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={5}
      mdOffset={1}>
      <Table
        className="table"
        columns={
          Array [
            Object {
              "key": "museumNo",
              "label": "Museumsnr"
            },
            Object {
              "key": "subNo",
              "label": "Unr"
            },
            Object {
              "key": "term",
              "label": "Term/artsnavn"
            }
          ]
        }
        data={Array []}
        defaultSort={false}
        defaultSortDescending={false}
        filterBy=""
        hideFilterInput={false}
        itemsPerPage={0}
        noDataText="Ingen objekter"
        sortBy={false}
        sortable={
          Array [
            "museumNumber",
            "subNumber",
            "term"
          ]
        } />
    </Col>
    <AddButton
      id="2"
      label="Legg til objekt"
      md={11}
      mdOffset={1} />
  </Form>
  <NewLine />
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={true}
    inline={false}
    style={
      Object {
        "paddingLeft": 20
      }
    }>
    <FormGroup
      bsClass="form-group">
      <Col
        bsClass="col"
        componentClass="div"
        md={12}>
        <h5>
          <strong>
            Personer tilknyttet analysen
          </strong>
        </h5>
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <FieldGroup
        id="responsible"
        label="Navn"
        md={1}
        onChange={[Function anonymous]}
        placeholder="Fornavn Etternavn"
        type="text"
        value="" />
      <LabelFormat
        label="Rolle"
        md={1} />
      <Col
        bsClass="col"
        componentClass="div"
        md={1}>
        <FormControl
          bsClass="form-control"
          componentClass="select"
          placeholder="Velg rolle">
          <option
            value="Velgsted">
            Velg rolle
          </option>
          <option
            value="other">
            ...
          </option>
        </FormControl>
      </Col>
      <AddButton
        id="3"
        label="Legg til person"
        md={2}
        mdOffset={0} />
    </FormGroup>
  </Form>
  <NewLine />
  <FormGroup
    bsClass="form-group">
    <LabelFormat
      label="Analysested"
      md={1} />
    <Col
      bsClass="col"
      componentClass="div"
      md={2}>
      <FormControl
        bsClass="form-control"
        componentClass="select"
        placeholder="Velg sted">
        <option
          value="Velgsted">
          Velg sted
        </option>
        <option
          value="other">
          ...
        </option>
      </FormControl>
    </Col>
  </FormGroup>
  <NewLine />
  <Well
    bsClass="well">
    <Form
      bsClass="form"
      componentClass="form"
      horizontal={true}
      inline={false}>
      <FormGroup
        bsClass="form-group">
        <LabelFormat
          label="Type analyse"
          md={1} />
        <Col
          bsClass="col"
          componentClass="div"
          md={2}>
          <FormControl
            bsClass="form-control"
            componentClass="select"
            onChange={[Function anonymous]}
            placeholder="Velg kategori">
            <option>
              Velg kategori
            </option>
            <option
              value="1bbf15cb-8348-4e66-99a4-bc314da57a42">
              3D-skanning, laser
            </option>
            <option
              value="b39399ab-aabd-4ebc-903b-adcf6876a364">
              3D-skanning, strukturert lys
            </option>
          </FormControl>
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group">
        <FieldGroup
          id="formControlsText"
          label="Ekstern kilde"
          md={1}
          onChange={[Function anonymous]}
          placeholder="http://www.lenke.no"
          type="text"
          value="" />
        <Col
          bsClass="col"
          componentClass="div"
          md={2}>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}>
            Lagre
          </Button>
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group">
        <FieldGroup
          id="formControlsText"
          label="Last opp fil"
          md={1}
          type="text" />
        <Col
          bsClass="col"
          componentClass="div"
          md={2}>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}>
            Bla gjennom
          </Button>
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group">
        <LabelFormat
          label="Kommentar / resultat"
          md={1} />
        <Col
          bsClass="col"
          componentClass="div"
          md={5}>
          <FormControl
            bsClass="form-control"
            componentClass="textarea"
            onChange={[Function anonymous]}
            placeholder=""
            value="" />
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group">
        <LabelFormat
          label="Klausulering"
          md={1} />
        <Col
          bsClass="col"
          componentClass="div"
          md={5}>
          <Radio
            bsClass="radio"
            disabled={false}
            inline={true}
            name="restrictions"
            onClick={[Function anonymous]}>
            Ja
          </Radio>
          <Radio
            bsClass="radio"
            defaultChecked={false}
            disabled={false}
            inline={true}
            name="restrictions"
            onClick={[Function anonymous]}>
            Nei
          </Radio>
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group">
        <Panel
          bsClass="panel"
          bsStyle="default"
          collapsible={true}
          defaultExpanded={false}
          expanded={true}
          style={
            Object {
              "backgroundColor": "#f5f5f5",
              "border": "none"
            }
          }>
          <FormGroup
            bsClass="form-group">
            <FieldGroup
              id="navn"
              label="Klausulert for"
              md={1}
              onChange={[Function anonymous]}
              placeholder="Fornavn Etternavn"
              type="text"
              value="" />
          </FormGroup>
          <FormGroup
            bsClass="form-group">
            <FieldGroup
              id="navn"
              label="Årsak til klausulering"
              md={1}
              onChange={[Function anonymous]}
              type="text"
              value="" />
          </FormGroup>
          <FormGroup
            bsClass="form-group">
            <FieldGroup
              id="Saksnummer"
              label="Saksnummer"
              md={1}
              onChange={[Function anonymous]}
              type="text"
              value={Array []} />
          </FormGroup>
          <FormGroup
            bsClass="form-group">
            <AddButton
              id="3"
              label="Legg til flere saksnummer"
              md={11}
              mdOffset={1} />
          </FormGroup>
          <FormGroup
            bsClass="form-group">
            <FieldGroup
              id="navn"
              label="Sluttdato"
              md={1}
              readOnly={true}
              type="text"
              value="" />
          </FormGroup>
          <FormGroup
            bsClass="form-group">
            <FieldGroup
              id="navn"
              label="Opphevet av"
              md={1}
              onChange={[Function anonymous]}
              placeholder="Fornavn Etternavn"
              type="text"
              value="" />
          </FormGroup>
          <FormGroup
            bsClass="form-group">
            <FieldGroup
              id="navn"
              label="Årsak til oppheving"
              md={1}
              onChange={[Function anonymous]}
              placeholder=""
              type="text"
              value="" />
          </FormGroup>
        </Panel>
      </FormGroup>
    </Form>
  </Well>
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={true}
    inline={false}
    style={
      Object {
        "paddingLeft": 20
      }
    }>
    <FormGroup
      bsClass="form-group"
      controlId="note"
      validationState={null}>
      <LabelFormat
        label="Kommentar til analysen"
        md={1} />
      <Col
        bsClass="col"
        componentClass="div"
        md={5}>
        <FormControl
          bsClass="form-control"
          className="note"
          componentClass="textarea"
          onChange={[Function anonymous]}
          placeholder="note"
          value="test note" />
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <LabelFormat
        label="Avslutt analyse"
        md={1} />
      <Col
        bsClass="col"
        componentClass="div"
        md={5}>
        <Radio
          bsClass="radio"
          disabled={false}
          inline={true}
          readOnly={true}>
          Ja
        </Radio>
        <Radio
          bsClass="radio"
          checked={true}
          disabled={false}
          inline={true}
          readOnly={true}>
          Nei
        </Radio>
      </Col>
    </FormGroup>
  </Form>
  <NewLine />
  <SaveCancel
    onClickSave={[Function onClickSave]} />
  <NewLine />
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={true}
    inline={false}>
    <FormGroup
      bsClass="form-group">
      <Col
        bsClass="col"
        componentClass="div"
        mdOffset={1}>
        <h5>
          <strong>
            Endringshistorikk
          </strong>
        </h5>
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <Col
        bsClass="col"
        componentClass="div"
        mdOffset={1}>
         - 
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <Col
        bsClass="col"
        componentClass="div"
        mdOffset={1}>
         - 
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <Col
        bsClass="col"
        componentClass="div"
        mdOffset={1}>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}>
          Se mer
        </Button>
      </Col>
    </FormGroup>
  </Form>
</div>
`;
